pkg/security/apis/security/validation/validation.go

```diff
@@ -77,17 +77,36 @@ func ValidateSecurityContextConstraints(scc *securityapi.SecurityContextConstrai
 			"required capabilities must be empty when all capabilities are allowed by a wildcard"))
 	}
 
-	if len(scc.Volumes) > 1 {
-		hasNone := false
+	hasFlexVolume := false
+	hasNoneVolume := false
+
+	if len(scc.Volumes) > 0 {
 		for _, fsType := range scc.Volumes {
 			if fsType == securityapi.FSTypeNone {
-				hasNone = true
-				break
+				hasNoneVolume = true
+
+			} else if fsType == securityapi.FSTypeFlexVolume || fsType == securityapi.FSTypeAll {
+				hasFlexVolume = true
 			}
 		}
-		if hasNone {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("volumes"), scc.Volumes,
-				"if 'none' is specified, no other values are allowed"))
+	}
+
+	if hasNoneVolume && len(scc.Volumes) > 1 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("volumes"), scc.Volumes,
+			"if 'none' is specified, no other values are allowed"))
+	}
+
+	if len(scc.AllowedFlexVolumes) > 0 {
+		if hasFlexVolume {
+			for idx, allowedFlexVolume := range scc.AllowedFlexVolumes {
+				if len(allowedFlexVolume.Driver) == 0 {
```

Comment:
I guess maybe you need validate the driver name format(we should have "openshift.com/foo" like driver name), if "foo" is specified instead of "openshift.io/foo", we should not pass the validation.